### PR TITLE
fix: [CDS-34198]: Handling inter process synchronisation for git operations

### PR DESCRIPTION
**Pull request by namantalaycha10**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/31314)

### CODE REVIEW

**Review by abosii-harness — DISMISSED** [review]
> Reviewed 8 of 8 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31314)* status: :shipit: complete! ✓

**Comment by namantalaycha10** [comment]
trigger authorcheck

**Review by namantalaycha10 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31314)* status: 2 file(s) to review. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[400-rest/src/main/java/software/wings/service/impl/yaml/GitClientImpl.java, line 765 at r3](https://reviewable.io/reviews/harness/harness-core/31314#-MxpCISF4YkOWHT8XCGm:-MxrAOT68omHRVMFkTVn:b6z3j4w) ([raw file](https://github.com/harness/harness-core/blob/25671ee8f2eaa83a5e891979b6be212779bb401c/400-rest/src/main/java/software/wings/service/impl/yaml/GitClientImpl.java#L765)):*
<details><summary><i>Previously, yogesh-chauhan (Yogesh Chauhan) wrote…</i></summary><blockquote>

how do we guarantee that that is the path where the git operation is going to take place ?
</blockquote></details>

hi @yogesh-chauhan, I tested both in CG and NG this is the path where .git/index.lock is present

**Review by shubham-in10se — CHANGES_REQUESTED** [review]
> Reviewed 6 of 15 files at r5, 4 of 4 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31314)* status: 5 file(s) to review. 2 change requests. 1. unresolved discussions. 0 of 1 approvals obtained. 

___
*[960-api-services/src/main/java/io/harness/git/Constants.java, line 26 at r6](https://reviewable.io/reviews/harness/harness-core/31314#-N-1xnCa3x7mImB-z7XX:-N-1xnCa3x7mImB-z7XY:bnv8csy) ([raw file](https://github.com/harness/harness-core/blob/337804dd921aa48f224a9b16c671e6a991bd0b4e/960-api-services/src/main/java/io/harness/git/Constants.java#L26)):*
> ```Java
>   String GIT_REPO_BASE_DIR = "./repository/${REPO_TYPE}/${ACCOUNT_ID}/${CONNECTOR_ID}/${REPO_NAME}/${REPO_URL_HASH}";
>   String REPOSITORY_GIT_FILE_DOWNLOADS = "./repository/gitFileDownloads";
>   String REPOSITORY_GIT_LOCK_DIR = "./repository/gitFileDownloads/%s";
> ```

create a `tmp` directory within the folder: 'gitFileDownloads' for the lock files.
Then, just append the lockFileName itself with the connectorId, instead creating a new directory every time.
___
*[960-api-services/src/main/java/io/harness/git/GitClientHelper.java, line 99 at r6](https://reviewable.io/reviews/harness/harness-core/31314#-N-1yLQq-XDmebw4lFCR:-N-1yLQq-XDmebw4lFCS:b-ia865j) ([raw file](https://github.com/harness/harness-core/blob/337804dd921aa48f224a9b16c671e6a991bd0b4e/960-api-services/src/main/java/io/harness/git/GitClientHelper.java#L99)):*
> ```Java
>             @Override
>             public Object load(String key) throws Exception {
>               return new File(format(REPOSITORY_GIT_FILE_LOCK, key));
> ```

change the return type to `File`

**Review by abosii-harness — CHANGES_REQUESTED** [review]
> Reviewed 8 of 15 files at r5, 4 of 4 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31314)* status: 3 file(s) to review. 3 change requests. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[960-api-services/src/main/java/io/harness/git/Constants.java, line 27 at r7](https://reviewable.io/reviews/harness/harness-core/31314#-N-MlOTe3WaCrTScsiak:-N-MlOTf3mrrvB7tE5C0:b-qli8o6) ([raw file](https://github.com/harness/harness-core/blob/7ed7c245405b18ed444c69ab75e293aaa1ab5bc0/960-api-services/src/main/java/io/harness/git/Constants.java#L27)):*
> ```Java
>   String REPOSITORY_GIT_FILE_DOWNLOADS = "./repository/gitFileDownloads";
>   String REPOSITORY_GIT_LOCK_DIR = "./repository/gitFileDownloads/locksTmpDir";
>   String REPOSITORY_GIT_FILE_LOCK = "./repository/gitFileDownloads/locksTmpDir/interProcessFileLock_%s";
> ```

nit: It can be REPOSITORY_GIT_FILE_LOCK  = REPOSITORY_GIT_LOCK_DIR + "interProcessFileLock_%s"
and I feel the naming is too "verbose". This is not a temporary directory and suffix Dir is not necessary. The same with file name, both can be just locks and lock_%s. And maybe we can make it hidden (use prefix . in locks directory) to avoid someone manually delete it by accident 
___
*[960-api-services/src/main/java/io/harness/git/GitClientHelper.java, line 99 at r6](https://reviewable.io/reviews/harness/harness-core/31314#-N-1yLQq-XDmebw4lFCR:-N-MmXln3GOEqoDoTZzM:b-ihq3tl) ([raw file](https://github.com/harness/harness-core/blob/337804dd921aa48f224a9b16c671e6a991bd0b4e/960-api-services/src/main/java/io/harness/git/GitClientHelper.java#L99)):*
<details><summary><i>Previously, shubham-in10se (Shubham Maheshwari) wrote…</i></summary><blockquote>

change the return type to `File`
</blockquote></details>

nit: In this case, parameter can be changed from Object to File 
___
*[960-api-services/src/main/java/io/harness/git/GitClientHelper.java, line 241 at r7](https://reviewable.io/reviews/harness/harness-core/31314#-N-MqaVl9yv2fJBM7IXz:-N-MqaVm59i0uURevUTj:b-47huex) ([raw file](https://github.com/harness/harness-core/blob/7ed7c245405b18ed444c69ab75e293aaa1ab5bc0/960-api-services/src/main/java/io/harness/git/GitClientHelper.java#L241)):*
> ```Java
>   }
> 
>   Object getLockObject(String id) {
> ```

We can return here File 
___
*[960-api-services/src/main/java/io/harness/git/GitClientHelper.java, line 243 at r7](https://reviewable.io/reviews/harness/harness-core/31314#-N-MmzsiADDIGfakf_lS:-N-MmzsiADDIGfakf_lT:bsmelmy) ([raw file](https://github.com/harness/harness-core/blob/7ed7c245405b18ed444c69ab75e293aaa1ab5bc0/960-api-services/src/main/java/io/harness/git/GitClientHelper.java#L243)):*
> ```Java
>   Object getLockObject(String id) {
>     try {
>       createDirectoryIfDoesNotExist(REPOSITORY_GIT_LOCK_DIR);
> ```

Is it safe? What if 2 threads/processes will try to create this directory at the same time?
Since this is a static directory maybe we can move this to static block in class? So it will be always created at class load
___
*[960-api-services/src/main/java/io/harness/git/GitClientHelper.java, line 245 at r7](https://reviewable.io/reviews/harness/harness-core/31314#-N-MnGxi6g_4wMDl-7ex:-N-MnGxi6g_4wMDl-7ey:b-4u4787) ([raw file](https://github.com/harness/harness-core/blob/7ed7c245405b18ed444c69ab75e293aaa1ab5bc0/960-api-services/src/main/java/io/harness/git/GitClientHelper.java#L245)):*
> ```Java
>       createDirectoryIfDoesNotExist(REPOSITORY_GIT_LOCK_DIR);
>       File file = new File(format(REPOSITORY_GIT_FILE_LOCK, id));
>       file.createNewFile();
> ```

why we're always trying to create a new file? Shouldn't we move this to CacheLoader so it will be created only if key is missing? Documentation mentioning that this is done atomically, I feel this atomicity is coming at cost and can be a performance issue if we will try always to create a new file at load. 
___
*[960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java, line 876 at r7](https://reviewable.io/reviews/harness/harness-core/31314#-N-MpxJ31jDueN7jskdU:-N-MpxJ31jDueN7jskdV:b-g0o87a) ([raw file](https://github.com/harness/harness-core/blob/7ed7c245405b18ed444c69ab75e293aaa1ab5bc0/960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java#L876)):*
> ```Java
> 
>     synchronized (gitClientHelper.getLockObject(gitConnectorId)) {
>       final File interProcessLockFile = (File) gitClientHelper.getLockObject(gitConnectorId);
> ```

let's move this before synchronized method and do synchronized on (interProcessLockFile)
nit: the name can be just lockFile
___
*[960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java, line 946 at r7](https://reviewable.io/reviews/harness/harness-core/31314#-N-MqZ-k9UgfZOH9LA4T:-N-MqZ-k9UgfZOH9LA4U:b-g0o87a) ([raw file](https://github.com/harness/harness-core/blob/7ed7c245405b18ed444c69ab75e293aaa1ab5bc0/960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java#L946)):*
> ```Java
>     validateRequiredArgs(request);
>     synchronized (gitClientHelper.getLockObject(request.getConnectorId())) {
>       File interProcessLockFile = (File) gitClientHelper.getLockObject(request.getConnectorId());
> ```

let's move this before synchronized method and do synchronized on (interProcessLockFile)
nit: the name can be just lockFile
___
*[960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java, line 1024 at r7](https://reviewable.io/reviews/harness/harness-core/31314#-N-MqfKJ6PfTkMmVUOuc:-N-MqfKJ6PfTkMmVUOud:b-g0o87a) ([raw file](https://github.com/harness/harness-core/blob/7ed7c245405b18ed444c69ab75e293aaa1ab5bc0/960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java#L1024)):*
> ```Java
> 
>     synchronized (gitClientHelper.getLockObject(request.getConnectorId())) {
>       final File interProcessLockFile = (File) gitClientHelper.getLockObject(request.getConnectorId());
> ```

let's move this before synchronized method and do synchronized on (interProcessLockFile)
nit: the name can be just lockFile
___
*[960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java, line 1100 at r7](https://reviewable.io/reviews/harness/harness-core/31314#-N-Mqzfq7YqOfphytjGY:-N-Mqzfq7YqOfphytjGZ:bl20s82) ([raw file](https://github.com/harness/harness-core/blob/7ed7c245405b18ed444c69ab75e293aaa1ab5bc0/960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java#L1100)):*
> ```Java
>   }
> 
>   // use this method wrapped in inter process file lock ro handle multiple delegate version
> ```

Does this mean that synchronized block is not required? 
___
*[960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java, line 1128 at r7](https://reviewable.io/reviews/harness/harness-core/31314#-N-MrALYBl4JWWhpfPDo:-N-MrALYBl4JWWhpfPDp:b-m9p3lr) ([raw file](https://github.com/harness/harness-core/blob/7ed7c245405b18ed444c69ab75e293aaa1ab5bc0/960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java#L1128)):*
> ```Java
>         }
>       } catch (Exception ex) {
>         // just to keep the exception handling behaviour same as before
> ```

Original code doesn't have catch block,  so why it is the same as before? 
nit: comment is not necessary here since the code doesn't have context "same as before" (or comment can be more explicit in code context)
___
*[400-rest/src/main/java/software/wings/service/impl/yaml/GitClientImpl.java, line 567 at r5](https://reviewable.io/reviews/harness/harness-core/31314#-N-MsLja7JPSAKT2l91E:-N-MsLjb2iGiyBRFIuqq:b-hacubo) ([raw file](https://github.com/harness/harness-core/blob/bebb58bca0fe935060f945f75b48075b6c0ebb4a/400-rest/src/main/java/software/wings/service/impl/yaml/GitClientImpl.java#L567)):*
> ```Java
>     validateRequiredArgsForFilesBetweenCommit(gitRequest.getOldCommitId(), gitRequest.getNewCommitId());
> 
>     synchronized (gitClientHelper.getLockObject(gitConnectorId)) {
> ```

Don't we need to adopt the same for GitClientImpl?

**Comment by tchaurasiya** [comment]
*[400-rest/src/main/java/software/wings/service/impl/yaml/GitClientImpl.java, line 567 at r5](https://reviewable.io/reviews/harness/harness-core/31314#-N-MsLja7JPSAKT2l91E:-N-NIz4F5ek6G_JW-gHg:b-jimofx) ([raw file](https://github.com/harness/harness-core/blob/bebb58bca0fe935060f945f75b48075b6c0ebb4a/400-rest/src/main/java/software/wings/service/impl/yaml/GitClientImpl.java#L567)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Don't we need to adopt the same for GitClientImpl?
</blockquote></details>

Now done for GitClientImpl too.

**Comment by tchaurasiya** [comment]
*[960-api-services/src/main/java/io/harness/git/Constants.java, line 27 at r7](https://reviewable.io/reviews/harness/harness-core/31314#-N-MlOTe3WaCrTScsiak:-N-NJCvODmMlG5G-1WkT:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7ed7c245405b18ed444c69ab75e293aaa1ab5bc0/960-api-services/src/main/java/io/harness/git/Constants.java#L27)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

nit: It can be REPOSITORY_GIT_FILE_LOCK  = REPOSITORY_GIT_LOCK_DIR + "interProcessFileLock_%s"
and I feel the naming is too "verbose". This is not a temporary directory and suffix Dir is not necessary. The same with file name, both can be just locks and lock_%s. And maybe we can make it hidden (use prefix . in locks directory) to avoid someone manually delete it by accident 
</blockquote></details>

Done.

**Comment by tchaurasiya** [comment]
*[960-api-services/src/main/java/io/harness/git/GitClientHelper.java, line 99 at r6](https://reviewable.io/reviews/harness/harness-core/31314#-N-1yLQq-XDmebw4lFCR:-N-NJH4e7kvHdjff6h8o:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/337804dd921aa48f224a9b16c671e6a991bd0b4e/960-api-services/src/main/java/io/harness/git/GitClientHelper.java#L99)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

nit: In this case, parameter can be changed from Object to File 
</blockquote></details>

Done.

**Comment by tchaurasiya** [comment]
*[960-api-services/src/main/java/io/harness/git/GitClientHelper.java, line 241 at r7](https://reviewable.io/reviews/harness/harness-core/31314#-N-MqaVl9yv2fJBM7IXz:-N-NJJR37WTyEq9HYEmy:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7ed7c245405b18ed444c69ab75e293aaa1ab5bc0/960-api-services/src/main/java/io/harness/git/GitClientHelper.java#L241)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

We can return here File 
</blockquote></details>

Done.

**Comment by tchaurasiya** [comment]
*[960-api-services/src/main/java/io/harness/git/GitClientHelper.java, line 243 at r7](https://reviewable.io/reviews/harness/harness-core/31314#-N-MmzsiADDIGfakf_lS:-N-NJO1p-BPqUfbPDUzY:b-2biz7z) ([raw file](https://github.com/harness/harness-core/blob/7ed7c245405b18ed444c69ab75e293aaa1ab5bc0/960-api-services/src/main/java/io/harness/git/GitClientHelper.java#L243)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Is it safe? What if 2 threads/processes will try to create this directory at the same time?
Since this is a static directory maybe we can move this to static block in class? So it will be always created at class load
</blockquote></details>

Moved it to static block.

**Comment by tchaurasiya** [comment]
*[960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java, line 876 at r7](https://reviewable.io/reviews/harness/harness-core/31314#-N-MpxJ31jDueN7jskdU:-N-NJZAwDAZ_Q6gZLJnR:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7ed7c245405b18ed444c69ab75e293aaa1ab5bc0/960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java#L876)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

let's move this before synchronized method and do synchronized on (interProcessLockFile)
nit: the name can be just lockFile
</blockquote></details>

Done.

**Comment by tchaurasiya** [comment]
*[960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java, line 1128 at r7](https://reviewable.io/reviews/harness/harness-core/31314#-N-MrALYBl4JWWhpfPDo:-N-NJcNf0xoJZDLMIENa:bwjao9z) ([raw file](https://github.com/harness/harness-core/blob/7ed7c245405b18ed444c69ab75e293aaa1ab5bc0/960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java#L1128)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

Original code doesn't have catch block,  so why it is the same as before? 
nit: comment is not necessary here since the code doesn't have context "same as before" (or comment can be more explicit in code context)
</blockquote></details>

Removed un used try block

**Comment by tchaurasiya** [comment]
*[960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java, line 946 at r7](https://reviewable.io/reviews/harness/harness-core/31314#-N-MqZ-k9UgfZOH9LA4T:-N-NKRO7Et4yf0jmmjSC:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7ed7c245405b18ed444c69ab75e293aaa1ab5bc0/960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java#L946)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

let's move this before synchronized method and do synchronized on (interProcessLockFile)
nit: the name can be just lockFile
</blockquote></details>

Done.

**Comment by tchaurasiya** [comment]
*[960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java, line 1024 at r7](https://reviewable.io/reviews/harness/harness-core/31314#-N-MqfKJ6PfTkMmVUOuc:-N-NKTctExrtinYsX_ZQ:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/7ed7c245405b18ed444c69ab75e293aaa1ab5bc0/960-api-services/src/main/java/io/harness/git/GitClientV2Impl.java#L1024)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

let's move this before synchronized method and do synchronized on (interProcessLockFile)
nit: the name can be just lockFile
</blockquote></details>

Done.

File: 400-rest/src/main/java/software/wings/service/impl/yaml/GitClientImpl.java [hunk 1]
```diff
@@ -749,8 +756,10 @@ public GitFetchFilesResult fetchFilesByPath(
     /*
      * ConnectorId is per gitConfig and will result in diff local path for repo
      * */
-    synchronized (gitClientHelper.getLockObject(gitConnectorId)) {
-      try {
+    File lockFile = gitClientHelper.getLockObject(gitConnectorId);
```
Comment (abhinav-harness) [inline 400-rest/src/main/java/software/wings/service/impl/yaml/GitClientImpl.java:759]: can you tell lockfile directory -> what is %s replaced with?

**Review by abosii-harness — APPROVED** [review]
> Reviewed 1 of 15 files at r5, 5 of 8 files at r8, 1 of 1 files at r9, 2 of 2 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31314)* status: 2 change requests. 3. unresolved discussions. 

___
*[400-rest/src/test/java/software/wings/service/impl/yaml/GitClientImplTest.java, line 225 at r8](https://reviewable.io/reviews/harness/harness-core/31314#-N-NbKS_0rXNQ6a0olZ8:-N-NbKS_0rXNQ6a0olZ9:b-idx9bi) ([raw file](https://github.com/harness/harness-core/blob/0b09ef5862c1a60b826024d938b6d6e03fb86fd5/400-rest/src/test/java/software/wings/service/impl/yaml/GitClientImplTest.java#L225)):*
> ```Java
>   private static final String CONNECTOR_ID = "CONNECTOR_ID";
> 
>   private static final LoadingCache<String, Object> cache =
> ```

is this used anywhere?

**Review by abosii-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31314)* status: 2 change requests. 3. unresolved discussions.

**Review by tchaurasiya — APPROVED** [review]
> Reviewed 7 of 15 files at r5, 5 of 8 files at r8, 1 of 1 files at r9, 2 of 2 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31314)* status: 2 change requests. 3. unresolved discussions.

**Review by tchaurasiya — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31314)* status: 2 change requests. 3. unresolved discussions.

**Comment by tchaurasiya** [comment]
*[400-rest/src/main/java/software/wings/service/impl/yaml/GitClientImpl.java, line 759 at r9](https://reviewable.io/reviews/harness/harness-core/31314#-MxjM1p0oNDQ-dWwTge4-r9-759:-N04zpzg6ZayQkFI1VOK:b-v09nze) ([raw file](https://github.com/harness/harness-core/blob/f47a6884b9882f3c02977d5c011d3d8bdcb18fff/400-rest/src/main/java/software/wings/service/impl/yaml/GitClientImpl.java#L759)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

can you tell lockfile directory -> what is %s replaced with?
</blockquote></details>

Its getting replaced by connectorId

**Review by tchaurasiya — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31314)* status: 2 change requests. 4. unresolved discussions. 

___
*[400-rest/src/main/java/software/wings/service/impl/yaml/GitClientImpl.java, line 765 at r3](https://reviewable.io/reviews/harness/harness-core/31314#-MxpCISF4YkOWHT8XCGm:-N04zvZE2FV26_CfQLhd:b-p9gvei) ([raw file](https://github.com/harness/harness-core/blob/25671ee8f2eaa83a5e891979b6be212779bb401c/400-rest/src/main/java/software/wings/service/impl/yaml/GitClientImpl.java#L765)):*
<details><summary><i>Previously, namantalaycha10 (Naman Talaycha) wrote…</i></summary><blockquote>

hi @yogesh-chauhan, I tested both in CG and NG this is the path where .git/index.lock is present  
</blockquote></details>

Verified in testing.

**Review by namantalaycha10 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31314)* status: 2 change requests. 4. unresolved discussions. 

___
*[400-rest/src/main/java/software/wings/service/impl/yaml/GitClientImpl.java, line 765 at r3](https://reviewable.io/reviews/harness/harness-core/31314#-MxpCISF4YkOWHT8XCGm:-N05-shY-QX3jgdsUIk_:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/25671ee8f2eaa83a5e891979b6be212779bb401c/400-rest/src/main/java/software/wings/service/impl/yaml/GitClientImpl.java#L765)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

Verified in testing.
</blockquote></details>

Done.
___
*[400-rest/src/main/java/software/wings/service/impl/yaml/GitClientImpl.java, line 759 at r9](https://reviewable.io/reviews/harness/harness-core/31314#-MxjM1p0oNDQ-dWwTge4-r9-759:-N05-rJP2Y_qzwzVxHPa:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/f47a6884b9882f3c02977d5c011d3d8bdcb18fff/400-rest/src/main/java/software/wings/service/impl/yaml/GitClientImpl.java#L759)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

Its getting replaced by connectorId
</blockquote></details>

Done.
___
*[960-api-services/src/main/java/io/harness/git/Constants.java, line 26 at r6](https://reviewable.io/reviews/harness/harness-core/31314#-N-1xnCa3x7mImB-z7XX:-N05-trrD8KjlJzqpwqH:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/337804dd921aa48f224a9b16c671e6a991bd0b4e/960-api-services/src/main/java/io/harness/git/Constants.java#L26)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.
___
*[960-api-services/src/main/java/io/harness/git/GitClientHelper.java, line 99 at r6](https://reviewable.io/reviews/harness/harness-core/31314#-N-1yLQq-XDmebw4lFCR:-N05-v84CerTAa25jkXE:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/337804dd921aa48f224a9b16c671e6a991bd0b4e/960-api-services/src/main/java/io/harness/git/GitClientHelper.java#L99)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Done.

**Review by tchaurasiya — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31314)* status: 2 change requests. 4. unresolved discussions.

**Review by tchaurasiya — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31314)* status: 2 change requests. 4. unresolved discussions.

**Review by tchaurasiya — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31314)* status: 2 change requests. 4. unresolved discussions.

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 7 of 15 files at r5, 5 of 8 files at r8, 1 of 1 files at r9, 2 of 2 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/31314)* status: 2 change requests. 3. unresolved discussions.

**Pull request merged**